// @amplication/code-gen-types/src/code-gen-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeGeneratorVersionStrategy = exports.EnumResourceType = exports.EnumPreviewAccountType = exports.EnumPendingChangeOriginType = exports.EnumPendingChangeAction = exports.EnumGitProvider = exports.EnumBlockType = exports.EnumAuthProviderType = exports.EnumActionStepStatus = exports.ModuleMap = exports.EnumModuleDtoDecoratorType = exports.EnumModuleActionRestInputSource = exports.EnumModuleDtoPropertyType = exports.EnumModuleActionRestVerb = exports.EnumModuleActionGqlOperation = exports.EnumModuleDtoType = exports.EnumModuleActionType = exports.EnumMessagePatternConnectionOptions = exports.EnumEntityPermissionType = exports.EnumEntityAction = exports.EnumDataType = void 0;
const files_1 = require("./files");
var models_1 = require("./models");
Object.defineProperty(exports, "EnumDataType", { enumerable: true, get: function () { return models_1.EnumDataType; } });
Object.defineProperty(exports, "EnumEntityAction", { enumerable: true, get: function () { return models_1.EnumEntityAction; } });
Object.defineProperty(exports, "EnumEntityPermissionType", { enumerable: true, get: function () { return models_1.EnumEntityPermissionType; } });
Object.defineProperty(exports, "EnumMessagePatternConnectionOptions", { enumerable: true, get: function () { return models_1.EnumMessagePatternConnectionOptions; } });
Object.defineProperty(exports, "EnumModuleActionType", { enumerable: true, get: function () { return models_1.EnumModuleActionType; } });
Object.defineProperty(exports, "EnumModuleDtoType", { enumerable: true, get: function () { return models_1.EnumModuleDtoType; } });
Object.defineProperty(exports, "EnumModuleActionGqlOperation", { enumerable: true, get: function () { return models_1.EnumModuleActionGqlOperation; } });
Object.defineProperty(exports, "EnumModuleActionRestVerb", { enumerable: true, get: function () { return models_1.EnumModuleActionRestVerb; } });
Object.defineProperty(exports, "EnumModuleDtoPropertyType", { enumerable: true, get: function () { return models_1.EnumModuleDtoPropertyType; } });
Object.defineProperty(exports, "EnumModuleActionRestInputSource", { enumerable: true, get: function () { return models_1.EnumModuleActionRestInputSource; } });
var EnumModuleDtoDecoratorType;
(function (EnumModuleDtoDecoratorType) {
    EnumModuleDtoDecoratorType["ObjectType"] = "ObjectType";
    EnumModuleDtoDecoratorType["InputType"] = "InputType";
    EnumModuleDtoDecoratorType["ArgsType"] = "ArgsType";
})(EnumModuleDtoDecoratorType || (exports.EnumModuleDtoDecoratorType = EnumModuleDtoDecoratorType = {}));
/**
 * ModuleMap is a map of module paths to modules
 * @deprecated Use FileMap instead
 * @see FileMap
 */
class ModuleMap extends files_1.FileMap {
    constructor(logger) {
        super(logger);
    }
    /**
     * Replace all modules paths using a function
     * @param fn A function that receives a module path and returns a new path
     */
    replaceModulesPath(fn) {
        this.replaceFilesPath(fn);
    }
    /**
     * Replace all modules code using a function
     * @param fn A function that receives a module code and returns a new code
     */
    async replaceModulesCode(fn) {
        await this.replaceFilesCode(fn);
    }
    /**
     * @returns An array of modules
     */
    modules() {
        return Array.from(this.getAll());
    }
}
exports.ModuleMap = ModuleMap;
var models_2 = require("./models");
Object.defineProperty(exports, "EnumActionStepStatus", { enumerable: true, get: function () { return models_2.EnumActionStepStatus; } });
Object.defineProperty(exports, "EnumAuthProviderType", { enumerable: true, get: function () { return models_2.EnumAuthProviderType; } });
Object.defineProperty(exports, "EnumBlockType", { enumerable: true, get: function () { return models_2.EnumBlockType; } });
Object.defineProperty(exports, "EnumGitProvider", { enumerable: true, get: function () { return models_2.EnumGitProvider; } });
Object.defineProperty(exports, "EnumPendingChangeAction", { enumerable: true, get: function () { return models_2.EnumPendingChangeAction; } });
Object.defineProperty(exports, "EnumPendingChangeOriginType", { enumerable: true, get: function () { return models_2.EnumPendingChangeOriginType; } });
Object.defineProperty(exports, "EnumPreviewAccountType", { enumerable: true, get: function () { return models_2.EnumPreviewAccountType; } });
Object.defineProperty(exports, "EnumResourceType", { enumerable: true, get: function () { return models_2.EnumResourceType; } });
Object.defineProperty(exports, "CodeGeneratorVersionStrategy", { enumerable: true, get: function () { return models_2.CodeGeneratorVersionStrategy; } });
