// @amplication/code-gen-types/src/dotnet-plugins.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DotnetEventNames = void 0;
var DotnetEventNames;
(function (DotnetEventNames) {
    DotnetEventNames["CreateEntityController"] = "CreateEntityController";
    DotnetEventNames["CreateEntityControllerBase"] = "CreateEntityControllerBase";
    DotnetEventNames["CreateEntityGrpcController"] = "CreateEntityGrpcController";
    DotnetEventNames["CreateEntityGrpcControllerBase"] = "CreateEntityGrpcControllerBase";
    DotnetEventNames["CreateEntityControllerSpec"] = "CreateEntityControllerSpec";
    DotnetEventNames["CreateServerAuth"] = "CreateServerAuth";
    DotnetEventNames["CreateAdminDotEnv"] = "CreateAdminDotEnv";
    DotnetEventNames["CreateAdminUI"] = "CreateAdminUI";
    DotnetEventNames["CreateServer"] = "CreateServer";
    DotnetEventNames["CreateServerAppModule"] = "CreateServerAppModule";
    DotnetEventNames["CreateProgramFile"] = "CreateProgramFile";
    DotnetEventNames["CreateServerGitIgnore"] = "CreateServerGitIgnore";
    DotnetEventNames["CreateAdminGitIgnore"] = "CreateAdminGitIgnore";
    DotnetEventNames["CreateMessageBroker"] = "CreateMessageBroker";
    DotnetEventNames["CreateMessageBrokerTopicsEnum"] = "CreateMessageBrokerTopicsEnum";
    DotnetEventNames["CreateMessageBrokerNestJSModule"] = "CreateMessageBrokerNestJSModule";
    DotnetEventNames["CreateMessageBrokerClientOptionsFactory"] = "CreateMessageBrokerClientOptionsFactory";
    DotnetEventNames["CreateMessageBrokerService"] = "CreateMessageBrokerService";
    DotnetEventNames["CreateMessageBrokerServiceBase"] = "CreateMessageBrokerServiceBase";
    DotnetEventNames["CreateEntityService"] = "CreateEntityService";
    DotnetEventNames["CreateEntityServiceBase"] = "CreateEntityServiceBase";
    DotnetEventNames["CreateServerDockerCompose"] = "CreateServerDockerCompose";
    DotnetEventNames["CreatePrismaSchema"] = "CreatePrismaSchema";
    DotnetEventNames["CreateServerCsproj"] = "CreateServerCsproj";
    DotnetEventNames["CreateServerAppsettings"] = "CreateServerAppsettings";
    DotnetEventNames["CreateAdminUIPackageJson"] = "CreateAdminUIPackageJson";
    DotnetEventNames["CreateEntityModule"] = "CreateEntityModule";
    DotnetEventNames["CreateEntityModuleBase"] = "CreateEntityModuleBase";
    DotnetEventNames["CreateEntityResolver"] = "CreateEntityResolver";
    DotnetEventNames["CreateEntityResolverBase"] = "CreateEntityResolverBase";
    DotnetEventNames["CreateSwagger"] = "CreateSwagger";
    DotnetEventNames["CreateSeed"] = "CreateSeed";
    DotnetEventNames["CreateEntityControllerToManyRelationMethods"] = "CreateEntityControllerToManyRelationMethods";
    DotnetEventNames["CreateEntityGrpcControllerToManyRelationMethods"] = "createEntityGrpcControllerToManyRelationMethods";
    DotnetEventNames["CreateEntityResolverToManyRelationMethods"] = "CreateEntityResolverToManyRelationMethods";
    DotnetEventNames["CreateEntityResolverToOneRelationMethods"] = "CreateEntityResolverToOneRelationMethods";
    DotnetEventNames["CreateDTOs"] = "CreateDTOs";
    DotnetEventNames["LoadStaticFiles"] = "LoadStaticFiles";
    DotnetEventNames["CreateConnectMicroservices"] = "CreateConnectMicroservices";
    /**
     * Event that will allow plugins to add secrets references to the SecretsKeyNames enum
     */
    DotnetEventNames["CreateServerSecretsManager"] = "CreateServerSecretsManager";
    DotnetEventNames["CreateEntityInterface"] = "CreateEntityInterface";
    DotnetEventNames["CreateEntityExtensions"] = "CreateEntityExtensions";
    DotnetEventNames["CreateEntityModel"] = "CreateEntityModel";
    DotnetEventNames["CreateResourceDbContextFile"] = "CreateResourceDbContextFile";
    DotnetEventNames["CreateSeedDevelopmentDataFile"] = "CreateSeedDevelopmentDataFile";
    DotnetEventNames["CreateControllerModuleFile"] = "CreateControllerModuleFile";
    DotnetEventNames["CreateControllerBaseModuleFile"] = "CreateControllerBaseModuleFile";
})(DotnetEventNames || (exports.DotnetEventNames = DotnetEventNames = {}));
