// prisma-schema-dsl-types/dist/types.js
"use strict";
exports.__esModule = true;
exports.FieldKind = exports.isCallExpression = exports.DB_GENERATED = exports.UUID = exports.CUID = exports.NOW = exports.AUTO_INCREMENT = exports.ReferentialActions = exports.ScalarType = exports.isDataSourceURLEnv = exports.DataSourceProvider = void 0;
var DataSourceProvider;
(function (DataSourceProvider) {
    DataSourceProvider["PostgreSQL"] = "postgresql";
    DataSourceProvider["MySQL"] = "mysql";
    DataSourceProvider["SQLite"] = "sqlite";
    DataSourceProvider["MongoDB"] = "mongodb";
    DataSourceProvider["MSSQLServer"] = "sqlserver";
})(DataSourceProvider = exports.DataSourceProvider || (exports.DataSourceProvider = {}));
function isDataSourceURLEnv(url) {
    return url.name !== undefined;
}
exports.isDataSourceURLEnv = isDataSourceURLEnv;
var ScalarType;
(function (ScalarType) {
    ScalarType["String"] = "String";
    ScalarType["Boolean"] = "Boolean";
    ScalarType["Int"] = "Int";
    ScalarType["BigInt"] = "BigInt";
    ScalarType["Float"] = "Float";
    ScalarType["Decimal"] = "Decimal";
    ScalarType["DateTime"] = "DateTime";
    ScalarType["Json"] = "Json";
})(ScalarType = exports.ScalarType || (exports.ScalarType = {}));
var ReferentialActions;
(function (ReferentialActions) {
    ReferentialActions["NONE"] = "NONE";
    ReferentialActions["Cascade"] = "Cascade";
    ReferentialActions["Restrict"] = "Restrict";
    ReferentialActions["NoAction"] = "NoAction";
    ReferentialActions["SetNull"] = "SetNull";
    ReferentialActions["SetDefault"] = "SetDefault";
})(ReferentialActions = exports.ReferentialActions || (exports.ReferentialActions = {}));
exports.AUTO_INCREMENT = "autoincrement";
exports.NOW = "now";
exports.CUID = "cuid";
exports.UUID = "uuid";
exports.DB_GENERATED = "dbgenerated";
function isCallExpression(object) {
    return object.callee !== undefined;
}
exports.isCallExpression = isCallExpression;
var FieldKind;
(function (FieldKind) {
    FieldKind["Scalar"] = "scalar";
    FieldKind["Object"] = "object";
})(FieldKind = exports.FieldKind || (exports.FieldKind = {}));
