// @amplication/csharp-ast/src/lib/utils/csharp-support.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Types = void 0;
exports.class_ = class_;
exports.annotation = annotation;
exports.classReference = classReference;
exports.genericClassReference = genericClassReference;
exports.instantiateClass = instantiateClass;
exports.invokeMethod = invokeMethod;
exports.coreClassReference = coreClassReference;
exports.codeblock = codeblock;
exports.field = field;
exports.method = method;
exports.parameter = parameter;
exports.interface_ = interface_;
exports.enum_ = enum_;
exports.dictionary = dictionary;
exports.programClass = programClass;
const ast_1 = require("../ast");
const GenericClassReference_1 = require("../ast/GenericClassReference");
function class_(args) {
    return new ast_1.Class(args);
}
function annotation(args) {
    return new ast_1.Annotation(args);
}
function classReference(args) {
    return new ast_1.ClassReference(args);
}
function genericClassReference(args) {
    return new GenericClassReference_1.GenericClassReference(args);
}
function instantiateClass(args) {
    return new ast_1.ClassInstantiation(args);
}
function invokeMethod(args) {
    return new ast_1.MethodInvocation(args);
}
function coreClassReference(args) {
    return new ast_1.CoreClassReference(args);
}
function codeblock(args) {
    return new ast_1.CodeBlock(args);
}
function field(args) {
    return new ast_1.Field(args);
}
function method(args) {
    return new ast_1.Method(args);
}
function parameter(args) {
    return new ast_1.Parameter(args);
}
function interface_(args) {
    return new ast_1.Interface(args);
}
function enum_(args) {
    return new ast_1.Enum(args);
}
function dictionary(args) {
    return new ast_1.Dictionary(args);
}
function programClass(args) {
    return new ast_1.ProgramClass(args);
}
exports.Types = ast_1.Type;
