// @amplication/code-gen-types/src/schemas/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.json = exports.file = exports.roles = exports.username = exports.password = exports.geographicLocation = exports.updatedAt = exports.createdAt = exports.id = exports.boolean = exports.multiSelectOptionSet = exports.optionSet = exports.lookup = exports.decimalNumber = exports.dateTime = exports.wholeNumber = exports.email = exports.multiLineText = exports.singleLineText = void 0;
var singleLineText_json_1 = require("./singleLineText.json");
Object.defineProperty(exports, "singleLineText", { enumerable: true, get: function () { return __importDefault(singleLineText_json_1).default; } });
var multiLineText_json_1 = require("./multiLineText.json");
Object.defineProperty(exports, "multiLineText", { enumerable: true, get: function () { return __importDefault(multiLineText_json_1).default; } });
var email_json_1 = require("./email.json");
Object.defineProperty(exports, "email", { enumerable: true, get: function () { return __importDefault(email_json_1).default; } });
var wholeNumber_json_1 = require("./wholeNumber.json");
Object.defineProperty(exports, "wholeNumber", { enumerable: true, get: function () { return __importDefault(wholeNumber_json_1).default; } });
var dateTime_json_1 = require("./dateTime.json");
Object.defineProperty(exports, "dateTime", { enumerable: true, get: function () { return __importDefault(dateTime_json_1).default; } });
var decimalNumber_json_1 = require("./decimalNumber.json");
Object.defineProperty(exports, "decimalNumber", { enumerable: true, get: function () { return __importDefault(decimalNumber_json_1).default; } });
var lookup_json_1 = require("./lookup.json");
Object.defineProperty(exports, "lookup", { enumerable: true, get: function () { return __importDefault(lookup_json_1).default; } });
var optionSet_json_1 = require("./optionSet.json");
Object.defineProperty(exports, "optionSet", { enumerable: true, get: function () { return __importDefault(optionSet_json_1).default; } });
var multiSelectOptionSet_json_1 = require("./multiSelectOptionSet.json");
Object.defineProperty(exports, "multiSelectOptionSet", { enumerable: true, get: function () { return __importDefault(multiSelectOptionSet_json_1).default; } });
var boolean_json_1 = require("./boolean.json");
Object.defineProperty(exports, "boolean", { enumerable: true, get: function () { return __importDefault(boolean_json_1).default; } });
var id_json_1 = require("./id.json");
Object.defineProperty(exports, "id", { enumerable: true, get: function () { return __importDefault(id_json_1).default; } });
var createdAt_json_1 = require("./createdAt.json");
Object.defineProperty(exports, "createdAt", { enumerable: true, get: function () { return __importDefault(createdAt_json_1).default; } });
var updatedAt_json_1 = require("./updatedAt.json");
Object.defineProperty(exports, "updatedAt", { enumerable: true, get: function () { return __importDefault(updatedAt_json_1).default; } });
var geographicLocation_json_1 = require("./geographicLocation.json");
Object.defineProperty(exports, "geographicLocation", { enumerable: true, get: function () { return __importDefault(geographicLocation_json_1).default; } });
var password_json_1 = require("./password.json");
Object.defineProperty(exports, "password", { enumerable: true, get: function () { return __importDefault(password_json_1).default; } });
var username_json_1 = require("./username.json");
Object.defineProperty(exports, "username", { enumerable: true, get: function () { return __importDefault(username_json_1).default; } });
var roles_json_1 = require("./roles.json");
Object.defineProperty(exports, "roles", { enumerable: true, get: function () { return __importDefault(roles_json_1).default; } });
var file_json_1 = require("./file.json");
Object.defineProperty(exports, "file", { enumerable: true, get: function () { return __importDefault(file_json_1).default; } });
var json_json_1 = require("./json.json");
Object.defineProperty(exports, "json", { enumerable: true, get: function () { return __importDefault(json_json_1).default; } });
