// @amplication/csharp-ast/src/lib/ast/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Solution = exports.ProgramClass = exports.Dictionary = exports.CoreClassReference = exports.MethodInvocation = exports.ClassInstantiation = exports.Interface = exports.Type = exports.Parameter = exports.MethodType = exports.Method = exports.Field = exports.Enum = exports.dependencies = exports.CodeBlock = exports.GenericClassReference = exports.ClassReference = exports.Class = exports.Annotation = void 0;
const tslib_1 = require("tslib");
var Annotation_1 = require("./Annotation");
Object.defineProperty(exports, "Annotation", { enumerable: true, get: function () { return Annotation_1.Annotation; } });
var Class_1 = require("./Class");
Object.defineProperty(exports, "Class", { enumerable: true, get: function () { return Class_1.Class; } });
var ClassReference_1 = require("./ClassReference");
Object.defineProperty(exports, "ClassReference", { enumerable: true, get: function () { return ClassReference_1.ClassReference; } });
var GenericClassReference_1 = require("./GenericClassReference");
Object.defineProperty(exports, "GenericClassReference", { enumerable: true, get: function () { return GenericClassReference_1.GenericClassReference; } });
var CodeBlock_1 = require("./CodeBlock");
Object.defineProperty(exports, "CodeBlock", { enumerable: true, get: function () { return CodeBlock_1.CodeBlock; } });
exports.dependencies = tslib_1.__importStar(require("./dependencies"));
var Enum_1 = require("./Enum");
Object.defineProperty(exports, "Enum", { enumerable: true, get: function () { return Enum_1.Enum; } });
var Field_1 = require("./Field");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return Field_1.Field; } });
var Method_1 = require("./Method");
Object.defineProperty(exports, "Method", { enumerable: true, get: function () { return Method_1.Method; } });
Object.defineProperty(exports, "MethodType", { enumerable: true, get: function () { return Method_1.MethodType; } });
var Parameter_1 = require("./Parameter");
Object.defineProperty(exports, "Parameter", { enumerable: true, get: function () { return Parameter_1.Parameter; } });
var Type_1 = require("./Type");
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return Type_1.Type; } });
var Interface_1 = require("./Interface");
Object.defineProperty(exports, "Interface", { enumerable: true, get: function () { return Interface_1.Interface; } });
var ClassInstantiation_1 = require("./ClassInstantiation");
Object.defineProperty(exports, "ClassInstantiation", { enumerable: true, get: function () { return ClassInstantiation_1.ClassInstantiation; } });
var MethodInvocation_1 = require("./MethodInvocation");
Object.defineProperty(exports, "MethodInvocation", { enumerable: true, get: function () { return MethodInvocation_1.MethodInvocation; } });
var CoreClassReference_1 = require("./CoreClassReference");
Object.defineProperty(exports, "CoreClassReference", { enumerable: true, get: function () { return CoreClassReference_1.CoreClassReference; } });
var Dictionary_1 = require("./Dictionary");
Object.defineProperty(exports, "Dictionary", { enumerable: true, get: function () { return Dictionary_1.Dictionary; } });
var ProgramClass_1 = require("./ProgramClass");
Object.defineProperty(exports, "ProgramClass", { enumerable: true, get: function () { return ProgramClass_1.ProgramClass; } });
var SolutionFile_1 = require("./SolutionFile");
Object.defineProperty(exports, "Solution", { enumerable: true, get: function () { return SolutionFile_1.Solution; } });
