// @amplication/csharp-ast/src/lib/ast/Parameter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Parameter = void 0;
const AstNode_1 = require("../core/AstNode");
class Parameter extends AstNode_1.AstNode {
    name;
    docs;
    initializer;
    annotations;
    splitAnnotations;
    type;
    constructor({ name, type, docs, initializer, annotations, splitAnnotations, }) {
        super();
        this.name = name;
        this.type = type;
        this.docs = docs;
        this.annotations = annotations ?? [];
        this.splitAnnotations = splitAnnotations ?? true;
    }
    write(writer) {
        if (this.annotations && this.annotations.length > 0) {
            !this.splitAnnotations && writer.write("[");
            this.annotations.forEach((annotation, index) => {
                if (this.splitAnnotations) {
                    writer.write("[");
                    annotation.write(writer);
                    writer.write("]");
                    writer.newLine();
                }
                else {
                    annotation.write(writer);
                    if (index < (this.annotations ? this.annotations.length : 0) - 1) {
                        writer.write(", ");
                    }
                }
            });
            !this.splitAnnotations && writer.write("]");
            writer.writeNewLineIfLastLineNot();
        }
        this.type.write(writer);
        writer.write(` ${this.name}`);
        if (this.initializer != null) {
            writer.write(` = ${this.initializer}`);
        }
    }
}
exports.Parameter = Parameter;
